config/v1/types_ingress.go

```diff
@@ -152,8 +153,64 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is optional and permitted only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network Load Balancer.
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations is a list of IDs for Elastic IP (EIP) addresses that
+	// are assigned to the Network Load Balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// See the documentation for the IngressController spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations
+	// API field for limitations on the use of Elastic IP addresses in the context of IngressControllers.
+	// This field only specifies EIPs for the default IngressController. It is not a default value
+	// for all IngressControllers.
+	// If the default IngressController specifies eipAllocations, then that value is used,
+	// and any eipAllocations value specified in this field is ignored.
+	//
+	// The following restrictions apply:
+	//
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	//
+	// Each entry in the eipAllocations list must be unique.
+	//
+	// A maximum of 10 IP allocations are permitted.
```

Comment:
Should we add the 2 additional restrictions (internal & single ingresscontroller) here? They also kinda of generally make sense to list here too.

My other comment is to keep the two Go Docs between IngressController and IngressConfig aligned. Deltas in the structure/order of the two docs might cause confusion and look a bit messy. If we align, it becomes more clear that these two fields are serving the very similar purpose, and it's easier to spot the difference between the two (Ingress Config should talk about defaulting).